src/adapter/graphql/tests/tests/test_gql_pagination.rs

```diff
@@ -0,0 +1,269 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use async_graphql::{value, EmptyMutation, EmptySubscription, Object, Schema};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[test_log::test(tokio::test)]
+async fn test_pagination() {
```

Comment:
Comment for readers: 

I had an error in the search code on pagination. For a long time I thought I found a bug in `page_based_connection!()`, but after writing these tests, I found a bug in my code.

Thus, I added tests on the working code :)